### PR TITLE
LimitRange e2e test improved.

### PR DESCRIPTION
**Pull request by kmrov**
**What this PR does / why we need it**: Improves the e2e test for LimitRange API by testing Update, Delete and Watch features.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by luxas** [comment]
/ok-to-test

> Also marks it as a Conformance test.

needs @kubernetes/sig-architecture-feature-requests (cc @timothysc) approval in a SIG meeting as well

File: test/e2e/scheduling/limit_range.go [hunk 1]
```diff
@@ -47,9 +51,34 @@ var _ = SIGDescribe("LimitRange", func() {
...
 			defaultLimit, defaultRequest,
 			maxLimitRequestRatio)
-		limitRange, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Create(limitRange)
+
+		By("Setting up watch")
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"name": limitRange.Name}))
+		options := metav1.ListOptions{LabelSelector: selector.String()}
+		limitRanges, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).List(options)
+		Expect(err).NotTo(HaveOccurred(), "failed to query for limitRanges")
+		Expect(len(limitRanges.Items)).To(Equal(0))
```
Comment (timothysc) [inline test/e2e/scheduling/limit_range.go]: I would warn here b/c we've seen a number of test conditions fail on loaded clusters due to waiting on cleanup conditions.  I realize that it shouldn't occur, but we've seen patterns of odd cleanup when running (N) overlapping test cases with namespaces names that are not completely unique.

File: test/e2e/scheduling/limit_range.go [hunk 2]
```diff
@@ -47,9 +51,34 @@ var _ = SIGDescribe("LimitRange", func() {
...
+		options = metav1.ListOptions{
+			LabelSelector:   selector.String(),
+			ResourceVersion: limitRanges.ListMeta.ResourceVersion,
+		}
+		w, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Watch(metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred(), "failed to set up watch")
+
+		By("Submitting a LimitRange")
+		limitRange, err = f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Create(limitRange)
```
Comment (timothysc) [inline test/e2e/scheduling/limit_range.go]: Update vs. Create based on previous value of list.

**Review by timothysc — APPROVED** [review]
/lgtm but we need @kubernetes/sig-architecture-pr-reviews to approve any conformance changes

@kmrov please add a release note though to callout any conformance additions.

**Comment by luxas** [comment]
We definitely need an associated issue here. Is there any tracking issue / doc for changes to the Conformance suite over time?

**Comment by bgrant0607** [comment]
@kmrov @timothysc 
Please do not mark this as conformance now. I'll put it on the SIG Arch. agenda.

File: test/e2e/scheduling/limit_range.go (hunk 1 above)
Comment (kmrov) [reply test/e2e/scheduling/limit_range.go]: What can you suggest to do to avoid the problems here?

File: test/e2e/scheduling/limit_range.go (hunk 2 above)
Comment (kmrov) [reply test/e2e/scheduling/limit_range.go]: Should Update at line 140 use a new LimitRange structure, not the old with changed limit value?

**Comment by dims** [comment]
Looks like the hold is for `conformance` annotation on the tests

**Comment by bgrant0607** [comment]
Yes, the hold was due to the conformance tag. Since that was removed, I removed the hold. 

Thanks for the test improvements.

**Comment by luxas** [comment]
@timothysc LGTM this for real as the Conformance tag was removed?

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by xiangpengzhao** [comment]
@timothysc a comment `/approve no-issue` is needed here :)
or
@kmrov you can create an issue for this PR and update the PR body accordingly to associate the PR with the issue (e.g., add `Fixes: #issue-number` to PR body), then the PR can get the `approved` label automatically (as @timothysc already commented `/approve`)

**Comment by spiffxp** [comment]
/status in-progress
though honestly at this point it doesn't really matter, since it's approved+lgtm+passing

**Comment by jberkus** [comment]
/priority critical-urgent

/remove-priority important-longterm

adjusting priorities for code freeze

**Pull request merged**